src/hooks/useApproveCurrencyTokenTransaction.ts

```diff
@@ -0,0 +1,19 @@
+import { useMemo } from 'react';
+
+import { useAppSelector } from '../redux/hooks';
+import { selectApprovals, SubmittedApproval } from '../redux/stores/transactions/transactionsSlice';
+
+const useApproveCurrencyTokenTransaction = (): SubmittedApproval | undefined => {
+  const { currencyToken } = useAppSelector(state => state.config);
+  const pendingTransactions = useAppSelector(selectApprovals);
+
+  return useMemo(() => {
+    if (!pendingTransactions.length) {
+      return undefined;
```

Comment:
Just return?

Reply:
Yes I want to return undefined if the `SubmittedApproval` does not exist.